File: client/src/app/shared/models/product-search-response.ts
export interface ProductSearchResponse {
  id: number;
  name: string;
  description: string;
  price: number;
  score: number;
  pictureUrl: string;
}


export interface ChatMessage {
  type: 'user' | 'bot';
  text?: string;
  products?: ProductSearchResponse[];
}

File: Back-End/Shared/Dtos/AiSearch/ProductSearchResponse.cs
namespace Shared.Dtos.AiSearch
{
    public class ProductSearchResponse
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public float Score { get; set; }
    }
}
